tests/test_messaging.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import helper
+from pythonwhat.Reporter import Reporter
+
+def message(output, patt):
+    return Reporter.to_html(patt) == output['message']
+
+def lines(output, s, e):
+    if s and e:
+        return output['column_start'] == s and output['column_end'] == e
+    else:
+        return True
+
+@pytest.mark.parametrize('stu, patt, cols, cole', [
+        ('', 'Did you call `round()`?', None, None),
+        ('round(1)', 'Check your call of `round()`. Did you specify the second argument?', 1, 8),
+        ('round(1, a)', 'Check your call of `round()`. Did you correctly specify the second argument? Rerunning it generated an error: `name \'a\' is not defined`.', 10, 10),
+        ('round(1, 2)', 'Check your call of `round()`. Did you correctly specify the second argument? Expected `3`, but got `2`.', 10, 10)
+])
+def test_check_function_pos(stu, patt, cols, cole):
+    output = helper.run({
+        'DC_CODE': stu,
+        'DC_SOLUTION': 'round(1, 3)',
+        'DC_SCT': 'Ex().check_function("round").check_args(1).has_equal_value()'
+    })
+    assert not output['correct']
+    assert message(output, patt)
+    assert lines(output, cols, cole)
+
+@pytest.mark.parametrize('stu, patt, cols, cole', [
+        ('round(1)', 'Check your call of `round()`. Did you specify the argument `ndigits`?', 1, 8),
+        ('round(1, a)', 'Check your call of `round()`. Did you correctly specify the argument `ndigits`? Rerunning it generated an error: `name \'a\' is not defined`.', 10, 10),
+        ('round(1, 2)', 'Check your call of `round()`. Did you correctly specify the argument `ndigits`? Expected `3`, but got `2`.', 10, 10)
+])
+def test_check_function_named(stu, patt, cols, cole):
+    output = helper.run({
+        'DC_CODE': stu,
+        'DC_SOLUTION': 'round(1, 3)',
+        'DC_SCT': 'Ex().check_function("round").check_args("ndigits").has_equal_value()'
+    })
+    assert not output['correct']
+    assert message(output, patt)
+    assert lines(output, cols, cole)
+
+@pytest.mark.parametrize('stu, patt, cols, cole', [
+    ('', 'Did you define the variable `x` without errors?', None, None),
+    ('x = 2', 'Did you correctly define the variable `x`? Expected `5`, but got `2`.', 1, 5)
+])
+def test_check_object(stu, patt, cols, cole):
+    output = helper.run({
+        'DC_CODE': stu,
+        'DC_SOLUTION': 'x = 5',
+        'DC_SCT': 'Ex().check_object("x").has_equal_value()'
+    })
+    assert not output['correct']
+    assert message(output, patt)
+    assert lines(output, cols, cole)
+
+@pytest.mark.parametrize('stu, patt', [
+    ('', 'The system wants to check the definition of `test()` but hasn\'t found it.'),
+    ('def test(a, b): return 1', 'Check the definition of `test()`. Calling `test(1, 2)` should return `3`, instead got `1`.'),
+    ('def test(a, b): return a + b', 'Check the definition of `test()`. Calling `test(1, 2)` should output `3`, instead got no output.'),
```

Comment:
Line 61 says "should return" and line 62 says "should output".

Reply:
Line 61 is failing on the `call(..., 'value')` function, so talks about what the function should return.
Line 62 is failing on the `call(..., 'output')` function, so talks about what the function should print out.
The messaging for output should be improved to be refer to print out. Thanks for flagging.